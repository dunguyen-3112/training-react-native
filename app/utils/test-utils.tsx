import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NavigationContainer } from '@react-navigation/native';

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
