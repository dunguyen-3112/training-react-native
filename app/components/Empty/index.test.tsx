import { render, screen } from '@utils/test-utils';
import React from 'react';
import Component from './index';

function setup() {
  const component = render(<Component />);
  expect(component).toMatchSnapshot();
}
describe('Component', () => {
  setup();
  it('Check Title', () => {
    const title = screen.getByText('No Results Found');
    expect(title).toBeTruthy();
  });
});
