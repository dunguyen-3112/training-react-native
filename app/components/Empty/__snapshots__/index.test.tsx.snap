// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 122,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/empty.png",
      }
    }
  />
  <Text
    onPress={[Function]}
    style={
      [
        {
          "borderColor": "#5C5C5C",
          "color": "#5C5C5C",
          "fontSize": 22,
          "fontWeight": "500",
          "textTransform": "none",
        },
        undefined,
        {
          "color": "#696969",
          "marginTop": 24,
        },
      ]
    }
  >
    No Results Found
  </Text>
  <Text
    onPress={[Function]}
    style={
      [
        {
          "borderColor": "#5C5C5C",
          "color": "#5C5C5C",
          "fontSize": 13,
          "fontWeight": "400",
          "textTransform": "none",
        },
        undefined,
        {
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Try search for a different keywork or
 tweek your search a little
  </Text>
</View>
`;
